src/SFML/Window/OSX/CursorImpl.mm

```diff
@@ -89,7 +89,10 @@
 
     switch (type)
     {
-        default: return false;
+        default:
+            if (m_cursor)
+                [m_cursor retain];
```

Comment:
If I understand the [memory model](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/MemoryMgmt/Articles/MemoryMgmt.html) of Obj-C correctly, this is not guaranteed to work, as the `release` at the top of the function, will allow the releasing of the cursor immediately, so one wouldn't be able to `retain` it here.

Reply:
Yeah, I think the correct approach here is to set the member to nil after releasing it, as if it's deallocated we end up with a dangly pointer, have got a PR incoming